### PR TITLE
[next] fix feature flag detection

### PR DESCRIPTION
**Pull request by styfle**
For feature flags, we use the string `1` for enabled and the string `0` for disabled.

So we need to check the value of the env var, not the presence of the env var.

### CODE REVIEW

**Review by feedthejim — DISMISSED** [review]

**Review by ijjk — DISMISSED** [review]

**Review by TooTallNate — APPROVED** [review]

**Review by trek — APPROVED** [review]

File: packages/next/src/server-build.ts [hunk 1]
```diff
@@ -313,7 +313,7 @@ export async function serverBuild({
 
     const useBundledServer =
       semver.gte(nextVersion, BUNDLED_SERVER_NEXT_VERSION) &&
-      process.env.VERCEL_NEXT_BUNDLED_SERVER;
+      process.env.VERCEL_NEXT_BUNDLED_SERVER === '1';
```
Comment (cb1kenobi) [inline packages/next/src/server-build.ts:316]: Or you can coerce it into a boolean :)

```suggestion
      ~~process.env.VERCEL_NEXT_BUNDLED_SERVER;
```

**Review by cb1kenobi — APPROVED** [review]

File: packages/next/src/server-build.ts (hunk 1 above)
Comment (styfle) [reply packages/next/src/server-build.ts:316]: @cb1kenobi No we can't, see the PR description

Comment (cb1kenobi) [reply packages/next/src/server-build.ts:316]: That's why I said `~~`:

```
> ~~null
0
> ~~undefined
0
> ~~''
0
> ~~false
0
> ~~0
0
> ~~'0'
0
> ~~true
1
> ~~1
1
> ~~'1'
1
```

Comment (TooTallNate) [reply packages/next/src/server-build.ts:316]: Too clever / unreadable IMO. I have no idea what that code does just by looking at it.

Comment (styfle) [reply packages/next/src/server-build.ts:316]: Very strange, thats not even coercing to a boolean, its coercing to a number.

Almost the same as `Number(x)`. But I think we should be explicit here here so I'm going to leave it.

I agree with Nate

Comment (Ethan-Arrowood) [reply packages/next/src/server-build.ts:316]: I didn't even know `~` is an operator lol

**Review by Ethan-Arrowood — APPROVED** [review]

**Pull request merged**